src/middleware/request-id/request-id.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * @module
+ * Request ID Middleware for Hono.
+ */
+
+import type { MiddlewareHandler } from '../../types'
+
+export type RequestIDVariables = {
+  requestID: string
+}
+
+export type RequesIDOptions = {
+  limitLength?: number
+  headerName?: string
+  generator?: () => string
+}
+
+/**
+ * Request ID Middleware for Hono.
+ *
+ * @param {object} options - Options for Request ID middleware.
+ * @param {number} [options.limitLength=255] - The maximum length of request id.
+ * If positive truncates the request id at the specified length.
+ * @param {string} [options.headerName=X-Request-Id] - The header name used in request id.
+ * @param {generator} [options.generator=crypto.randomUUID()] - The request id generation function.
+ *
+ * @returns {MiddlewareHandler} The middleware handler function.
+ *
+ * @example
+ * ```ts
+ * type Variables = RequestIDVariables
+ * const app = new Hono<{Variables: Variables}>()
+ *
+ * app.use(requestID())
+ * app.get('/', (c) => {
+ *   console.log(c.get('requestID')) // Debug
+ *   return c.text('Hello World!')
+ * })
+ * ```
+ */
+export const requestID = (options?: RequesIDOptions): MiddlewareHandler => {
+  const limitLength = options?.limitLength ?? 255
+  const headerName = options?.headerName ?? 'X-Request-Id'
+
+  return async function requestID(c, next) {
+    // If `headerName` is empty string, req.header will return the object
+    let requestId = headerName ? c.req.header(headerName) : undefined
```

Comment:
is this really needed?

since we already provide a check for this

```ts
const headerName = options?.headerName ?? 'X-Request-Id'
```

Reply:
> is this really needed?
> 
> since we already provide a check for this
> 
> ```ts
> const headerName = options?.headerName ?? 'X-Request-Id'
> ```


I haven't documented this yet, but it is for internal use only. For example, logging.
Specifying empty string means that the user expects no request ID to be present in  request/response.
And we need to check for empty string.

```ts
console.log("" ?? "string") // empty
console.log("" || "string") // string
console.log("" ? "A" : "B") // B
```

Reply:
Ah I see, good approach!

Reply:
You can close this comment